rcp/org.testeditor.rcp4.uatests/src/test/java/org/testeditor/rcp4/writetests/AutocompletionProposalTest.tcl

```diff
@@ -0,0 +1,140 @@
+	- Type "\n" into active editor
+	// then
+	- singleDashedInteraction = Contains editor "\n- do something \"1\""
+	- assert singleDashedInteraction
+
+* Given a prefixed of "- do some" _ assist provides only the ones limited to the ones matching that prefix
+
+	Component: ActiveEditor
+	// given
+	- Type "\n- do some" into active editor
+	// when
+	- hasDoSome = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	- hasDoSomeDiff = Has autocompletion proposal "do something different \"count: long\" - test step" when entering ""
+	- shouldNotHaveDoDiff = Has autocompletion proposal "do differently \"count: long\" - test step" when entering ""
+	// then
+	- assert hasDoSome
+	- assert hasDoSomeDiff
+	- assert !shouldNotHaveDoDiff
+
+* Given a content completion based on a prefix _ selecting the proposal inserts exactly the proposal and no superfluous dashes remain
+
+	Component: ActiveEditor
+	// given
+	// when
+	- Use auto completion with "" and select "do something different \"count: long\" - test step"
+	- Type "\n" into active editor
+	// then
+	- singleDashedInteractionPrefixed = Contains editor "\n- do something different \"1\""
+	- assert singleDashedInteractionPrefixed
+
+* Given a prefix ending in a space eg "- do something different " _ proposal will still work and be narrowed apropriately
+
+	Component: ActiveEditor
+	// given
+	- Type "\n- do something different " into active editor
+	// when
+	- hasDoSomeDiffEvenWithPrefixedSpace = Has autocompletion proposal "do something different \"count: long\" - test step" when entering ""
+	- shoudlNotHaveDoSomeAnymore = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	// then
+	- assert hasDoSomeDiffEvenWithPrefixedSpace
+	- assert !shoudlNotHaveDoSomeAnymore
+
+* Given an assignment _ make sure only templates that actually ^return something are available
+
+	Component: ActiveEditor
+	// given
+	- Type "\n\n- somevar = " into active editor
+	// when
+	- hasDoReturn = Has autocompletion proposal "doreturn something <element> - test step" when entering ""
+	- shouldNotHaveSomething = Has autocompletion proposal "do something \"count: long\" - test step" when entering ""
+	// then
+	- assert hasDoReturn
+	- assert !shouldNotHaveSomething
+	
+Setup:
+
+	// test editor project
+	Component: ProjectExplorer
+	- Execute menu item "New/Project..." in tree <ProjectTree>
+
+	// with gradle and web fixture
+	Component: NewProjectDialog
+	- Select element "Test-Editor Project" in tree <ProjectType>
+	- Click on <NextButton>
+	- Type "MyFirstWebProject" into <ProjectName>
+	- Click on <NextButton>
+	- Select value "Gradle" in combo box <BuildTool>
+	- Select element "Web Fixture" in list <AvailableFixturesList>
+	- Click on <AddFixtureButton>
+	- Click on <FinishButton>
+	- Wait for dialog "Progress Information" to popup and then close after at most "3" respectively "120" seconds
```

Comment:
So… wait for how long now? (I don't understand that sentence.)

Reply:
yeah, is a bit misleading, have no other compact solution that will order the parameters by importance. I'm using something in existence, though